src/hazelcore/HTyp.re

```diff
@@ -180,3 +175,8 @@ let join_all = (j: join, types: list(t)): option(t) => {
     }
   };
 };
+
+let is_unknown: t => bool =
+  fun
+  | Unknown(_) => true
+  | _ => false;
```

Comment:
if we're treating ModeSwitch as an unknown for consistency, then we should maybe treat it as an unknown everywhere (perhaps putting it back into Unknown as a provenance instead of a top-level thing)